frontend/views/components/Graphs/Bars.vue

```diff
@@ -1,48 +1,223 @@
 <template lang='pug'>
-.has-text-centered
-  p.c-description
-    strong.is-size-5.is-block.c-nowrap {{currency}} {{mincome}}
-    i18n min income
-
-  div(v-if='history.length')
-    .c-graph
-      span.c-graph-item(
-        v-for='(percentage, index) in history'
-        :key='`percentage-${index}`'
-        :style='{\
-          transform: `scaleY(${getHeight(percentage)})`,\
-          opacity: getOpacity(percentage)\
-        }'
-      )
+.c-chart-wrapper
+  svg(
+    width='100%'
+    :viewBox='`0 0 ${ratioX} ${ratioY}`'
+    preserveAspectRatio='xMidYMid meet'
+    aria-labelledby='title'
+    role='img'
+    ref='graph'
+  )
+    // Move graph origin to the middle
+    g(:transform='`translate(0,${middle})`')
+      // Animate using scale from the middle
+      g.g-animate(:style='`transform: scale3d(1,${ ready ? 1 : 0 },1)`')
+        g.graph-bar(
+          v-for='(member, index) in members'
+          :transform='positionX(index)'
+        )
+          // Total needed or total pledge bars
+          path(
+            v-for='(values, index) in [member.total, member.amount]'
+            :class='color(values, index)'
+            :d='roundedRect(values >= 0, 0, positionY(values), width, height(values), width > 15 ? 3 : 1)'
+          )
+
+      // Base with $0 on top of bars
+      line(x1='0' y1='0' :x2='ratioX' y2='0' stroke='#dbdbdb' stroke-width='1')
 
-    p.c-description
-      strong.is-block.c-nowrap
-        | + {{getTotal()}}
-        i18n distributed
-      i18n on the last
-      |  {{history.length}}
-      i18n months
+      // Surplus line on top of bars
+      g(
+        v-for='scaleLine in createScale'
+        :transform='`translate(0,${-scaleLine.position})`'
+      )
+        line.g-animate.g-animate-delay(
+          :style='`transform: scale3d(${ready ? 1 : 0},1,1)`'
+          x1='0'
+          y1='0'
+          :x2='ratioX'
+          y2='0'
+          stroke='#dbdbdb'
+          stroke-width='1'
+          stroke-dasharray='1'
+        )
 
-  p(v-else='') We are on the first month.
+  .tag.mincome(:style='middleTag') {{ base }}
+  .tag.g-animate-opacity.g-animate-delay(
+    v-for='scaleLine in createScale'
+    :style='{ opacity: ready ? 1 : 0, transform: "translate(0," + (-scaleLine.position + middle - labelPadding) + "px)" }'
+  ) {{ scaleLine.label }}
 </template>
 
 <script>
+import { mapGetters } from 'vuex'
+import { debounce } from '@utils/giLodash.js'
+import { TABLET } from '@view-utils/breakpoints.js'
+import currencies from '@view-utils/currencies.js'
+
 export default {
   name: 'Bars',
   props: {
-    history: Array,
-    currency: String,
-    mincome: Number
+    totals: Array,
+    members: Array
+  },
+  data: () => ({
+    ratioWidthPadding: 1.5,
+    ratioX: 526,
```

Comment:
526 seems useless here because right on first render `handleResize` is called and overrides `ratioX ` value.
Perhaps that's related to the weird flick mentioned on the other comment 🤔

Reply:
It is related, it's because the $ref of the element to calculate the max-width of the graph is available after the actual compilation of the graph. If we put null then we will have a error before the $ref is available.